### ISSUE TITLE
Macos signing does not work with Self-hosted runners

### ISSUE THREAD

**Issue by DavidGOrtega** [electron-userland/electron-builder#7844]
* **Electron-Builder Version**: 
24.8.0

* **Node Version**:
18.4.0

* **Electron Version**:
* **Electron Type (current, beta, nightly)**:
27.0.2

* **Target**: 
MacOS

Im launching a self-hosted runner to build my app. Here is the GH actions file:

```yaml

name: build
on:
  push:
    tags:
      - "v*"

jobs:
  build:
    runs-on: [self-hosted, M1]
    permissions:
      contents: write
    steps:
      - uses: actions/checkout@v2

      - uses: actions/setup-node@v2
        with:
          node-version: "18"

      - name: Build
        env:
          CSC_LINK: ${{ secrets.MACOS_CERT }}
          CSC_KEY_PASSWORD: ${{ secrets.MACOS_CERT_PASS }}
          NOT_APPLE_ID: ${{ secrets.APPLE_ID }}
          NOT_APPLE_ID_PASS: ${{ secrets.APPLE_ID_PASS }}
        run: |

          yarn install
          yarn run build-desktop

      - name: Release
        uses: softprops/action-gh-release@v1
        with:
          files: |
            packages/008desktop/build/008-desktop-universal.dmg
```

The cert is the secret `MACOS_CERT` stored as base64, with it Im generating the env variable `CSC_LINK` (Clarification just for non GHA users)
This workflow works perfectly with GH hosted runners. However in my self-hosted runner does not.

Seems that the workflow skips the sign

```sh

• skipped macOS application code signing  reason=cannot find valid "Developer ID Application" identity or custom non-Apple code signing certificate, it could cause some undefined behaviour, e.g. macOS localized description not visible, see https://electron.build/code-signing allIdentities=  1) *******EDITED************* "Developer ID Application: ************* EDITED **************" (CSSMERR_TP_NOT_TRUSTED)
     1 identities found
                                                Valid identities only
     0 valid identities found
```

the env variable is accesible.

**Comment by DavidGOrtega** [comment]
Following the code seems that the ENV var does not prevail?

```js
protected getCscLink(extraEnvName?: string | null): string | null | undefined {
    // allow to specify as empty string
    const envValue = chooseNotNull(extraEnvName == null ? null : process.env[extraEnvName], process.env.CSC_LINK)
    return chooseNotNull(chooseNotNull(this.info.config.cscLink, this.platformSpecificBuildOptions.cscLink), envValue)
  }
```

**Comment by DavidGOrtega** [comment]
Update:

`cscLink` property also works in the CI runner but not in the self-hosted runner.

**Comment by DavidGOrtega** [comment]
Setting  `CSC_IDENTITY_AUTO_DISCOVERY=false` behaves differently as stated in the docs. Its trowing an error

```sh
• skipped macOS application code signing  reason=, see https://electron.build/code-signing CSC_IDENTITY_AUTO_DISCOVERY=false
```

however docs [states](https://www.electron.build/code-signing.html#how-to-disable-code-signing-during-the-build-process-on-macos):

```
To disable Code Signing when building for macOS leave all the above vars unset except for CSC_IDENTITY_AUTO_DISCOVERY which needs to be set to false. This can be done by running export CSC_IDENTITY_AUTO_DISCOVERY=false.
```

clearly my vars `CSC_LINK` and `CSC_KEY_PASSWORD` vars are set so the logic or the docs might be wrong maybe? Another bug?

**Comment by DavidGOrtega** [comment]
I made it work, however electron-builder is not generating the cert on its own despite it creates the electron-builder-root-certs keychain. The runner has enough permissions to create the cert on its own as the GH hosted runner does in their MacOS machine, so Im not sure why it fails to deliver.

For now to make it work I had to setup the certs locally within the login keychain in the M1 machine where the self-hosted runner is operating.
